### DISCUSSION TITLE
Is it possible to show the terminal a ConPTY instance is interacting with?

### DISCUSSION THREAD

**Discussion by Unit2795** [microsoft/terminal#16791] [Q&A] [answered]
I connect a ConPTY instance to a Powershell process. Instead of directing the output from ConPTY to another Powershell terminal to show the output, is it possible to directly show the child Powershell terminal and watch as ConPTY interacts with it?

**Comment by zadjii-msft** [comment, marked as answer]
No...ish. Definitely not with the public APIs.

However, I'm pretty sure we had some tools in this repo for manually setting up conpty in a way that would let you see the conhost. I'm thinking of https://github.com/microsoft/terminal/blame/main/src/tools/vtpipeterm/main.cpp specifically. However, last I worked with that tool was probably 5 years ago so I'm not sure that even works with the current state of the console anymore.

**Reply by Unit2795** [reply]
Thanks, I'll take a look at this.

**Reply by Unit2795** [reply]
This works, many thanks! I think I could end up using something like this for debugging/auditing and interacting with automations as they take place.
